trestle/transforms/transformer_helper.py

```diff
@@ -14,12 +14,132 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Transformer helper functions."""
-
+from typing import Any
+from typing import Dict
 from typing import List
 
 from trestle.oscal.assessment_results import Observation
 from trestle.oscal.common import Property
 
+_segment_separator = '|'
+
+
+class PropertyAccounting():
+    """Property accounting class.
+
+    Help transformers do accounting.
+
+    > Each time a new record is processed the transformer calls count_group.
+    > For each attribute on that record, the transformer calls count_property.
+      - If the property already exactly exists, then its count is incremented.
+      - Otherwise, a new entry is made and the count for that property is set to 1.
+    > When the transformer wants to know if a property (name, value, ns, and class)
+      is common to all records for the group, is_common_property is employed to check
+      that the number of records in the group is equal to the number of duplicates
+      there are for the property. If equal, then the property is common.
+    """
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._group_map: Dict[str, int] = {}
+        self._property_map: Dict[str, Dict[str:int]] = {}
+
+    def count_group(self, group: str = None) -> None:
+        """Group accounting."""
+        if group not in self._group_map:
+            self._group_map[group] = 0
+        self._group_map[group] += 1
+
+    def count_property(
+        self, group: str = None, name: str = None, value: str = None, class_: str = None, ns: str = None
+    ) -> None:
+        """Property accounting."""
+        key = _segment_separator.join([str(name), str(value), str(class_), str(ns)])
+        if group not in self._property_map:
+            self._property_map[group] = {}
+        if key not in self._property_map[group]:
+            self._property_map[group][key] = 0
+        self._property_map[group][key] += 1
+
+    def is_common_property(
+        self, group: str = None, name: str = None, value: str = None, class_: str = None, ns: str = None
+    ) -> bool:
+        """Check for common property."""
+        rval = False
+        key = _segment_separator.join([str(name), str(value), str(class_), str(ns)])
+        if group in self._group_map and key in self._property_map[group]:
+            rval = self._group_map[group] == self._property_map[group][key]
+        return rval
+
+
+class PropertyManager():
```

Comment:
Shall we add tests for this and the above?

Reply:
Done. No bugs found this time ;-)